File: taskflow/src/controllers/AuthController.js
import { registerUserInFirebase, loginUserInFirebase, logoutFromFirebase, updateUserProfile } from "../models/UserModel";

export async function registerUser({ name, email, pass, pass2 }) {
  if (!name.trim()||!email.trim()||!pass||!pass2) return {ok:false,error:"Completa todos los campos."};
  if (pass!==pass2) return {ok:false,error:"Las contraseñas no coinciden."};
  if (pass.length<6) return {ok:false,error:"Mínimo 6 caracteres en la contraseña."};
  try { return {ok:true, user:await registerUserInFirebase(name.trim(),email.trim(),pass)}; }
  catch(e) { return {ok:false, error:firebaseErrorMessage(e.code)}; }
}

export async function loginUser(email, password) {
  if (!email.trim()||!password.trim()) return {ok:false,error:"Completa correo y contraseña."};
  try { return {ok:true, user:await loginUserInFirebase(email.trim(),password)}; }
  catch(e) { return {ok:false, error:firebaseErrorMessage(e.code)}; }
}

export async function logoutUser() { await logoutFromFirebase(); }

export async function saveUserPreferences(uid, city, interests) {
  await updateUserProfile(uid, { city, interests, needsOnboarding:false });
}

function firebaseErrorMessage(code) {
  const map = {
    "auth/email-already-in-use":   "Este correo ya está registrado.",
    "auth/invalid-email":          "El correo no es válido.",
    "auth/weak-password":          "La contraseña es muy débil.",
    "auth/user-not-found":         "No existe una cuenta con este correo.",
    "auth/wrong-password":         "Contraseña incorrecta.",
    "auth/invalid-credential":     "Correo o contraseña incorrectos.",
    "auth/too-many-requests":      "Demasiados intentos. Intenta más tarde.",
    "auth/network-request-failed": "Error de red. Revisa tu conexión.",
  };
  return map[code]??"Ocurrió un error. Intenta de nuevo.";
}

File: taskFlow kotlin/src/main/kotlin/com/taskflow/controllers/AuthController.kt
package com.taskflow.controllers

import com.taskflow.models.AuthResponse
import com.taskflow.models.LoginRequest
import com.taskflow.models.RegisterRequest
import com.taskflow.models.UpdateUserRequest
import com.taskflow.models.User
import com.taskflow.repository.UserRepository
import com.taskflow.service.AuthService

class AuthController {

    private val service  = AuthService()
    private val userRepo = UserRepository()

    suspend fun registrar(req: RegisterRequest): AuthResponse {
        if (req.name.isBlank() || req.email.isBlank() || req.password.isBlank() || req.password2.isBlank())
            return AuthResponse(ok = false, error = "Completa todos los campos.")
        if (req.password != req.password2)
            return AuthResponse(ok = false, error = "Las contraseñas no coinciden.")
        if (req.password.length < 6)
            return AuthResponse(ok = false, error = "Mínimo 6 caracteres en la contraseña.")
        return service.registrar(req.name.trim(), req.email.trim(), req.password)
    }

    suspend fun login(req: LoginRequest): AuthResponse {
        if (req.email.isBlank() || req.password.isBlank())
            return AuthResponse(ok = false, error = "Completa correo y contraseña.")
        return service.login(req.email.trim(), req.password)
    }

    suspend fun obtenerPerfil(uid: String): User? = userRepo.getById(uid)

    suspend fun actualizarPerfil(uid: String, req: UpdateUserRequest): Boolean {
        val updates = mutableMapOf<String, Any>()
        req.name?.let            { updates["name"]            = it.trim() }
        req.city?.let            { updates["city"]            = it }
        req.interests?.let       { updates["interests"]       = it }
        req.needsOnboarding?.let { updates["needsOnboarding"] = it }
        if (updates.isEmpty()) return false
        return userRepo.update(uid, updates)
    }
}
